### ISSUE TITLE
torch.Tensor.bag() should automatically implement bagging

### ISSUE THREAD

**Issue by aogara-ds** [pytorch/pytorch#82761]
### 🚀 The feature, motivation and pitch

Bootstrap aggregation, or bagging, is a common sampling technique in training ML models. For a dataset D of size N, bagging will randomly sample with replacement N data points from D. These bagged datasets can then be used to train an ensemble of models which will often outperform a single model trained on unbagged data (see Breiman, 1996). 

Tensor.bag(dim=0, n=1) should return a tensor of data points sampled randomly with replacement from the original tensor, and of the same size as that tensor. 

Dim refers the the dimension along which to randomly sample. 

N refers to the number of bagged datasets to return; if n>1, the tensor returned should include n independent bagged datasets of the same size as the original tensor. 

Out refers to what should be returned. The default is to return the bagged values. 'idx' should return the indices that would have been bagged, and 'both' should return a named tuple of both values and indices. 

Questions:
* Should the returned Tensor be a deep copy or simply a view of the original data? Views are sufficient if the data will not be transformed in the training process. But processing after bagging (e.g. normalizing the bagged dataset) would require a deep copy. Perhaps this should be an option, or two separate methods. 
* Is 'out' the correct way to handle this? What should be the default behavior, and what should the options be named?

### Alternatives

PyTorch could also implement the equivalent of numpy's np.random.choice(), which has an outstanding pull request as described in this issue: https://github.com/pytorch/pytorch/issues/16897
